daemon/media_socket.c

```diff
@@ -226,14 +224,140 @@ static GHashTable *__logical_intf_name_family_hash;
 static GHashTable *__intf_spec_addr_type_hash;
 static GQueue __preferred_lists_for_family[__SF_LAST];
 
+static unsigned int selection_index = 0;
```

Comment:
These 2 vars are also global and so need some synchronization. It looks like converting them to thread-local storage (using __thread) would be enough.